### PR TITLE
CAPBM: cluster manager unit tests

### PR DESCRIPTION
**Pull request by Jaakko-Os**

### CODE REVIEW

File: baremetal/cluster_manager_test.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	baremetalClusterName = "testBaremetalCluster"
+	namespaceName        = "testNameSpace"
+)
+
+type tcTest struct {
+	c             client.Client
+	BMCluster     *infrav1.BareMetalCluster
+	Cluster       *clusterv1.Cluster
+	ExpectSuccess bool
+}
```
Comment (maelk) [inline baremetal/cluster_manager_test.go]: this structure is unused

File: baremetal/cluster_manager_test.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+				}
+			} else {
+				err = clusterMgr.Create(context.TODO())
+				if err != nil {
+					if tc.ExpectSuccess {
+						t.Error(err)
+					} else { // FAILURE is expected
+						t.Logf("Create, expected result: %v", err)
+					}
+				}
```
Comment (maelk) [inline baremetal/cluster_manager_test.go]: this does not check if you were expecting a failure but did not get an error!

File: baremetal/cluster_manager_test.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+					if err != nil {
+						t.Error(err)
+					} else {
+						t.Logf("APIEndPoints %v", apiEndPoints)
+					}
+				} else { // FAILURE is expected
+					if err != nil {
+						t.Logf("APIEndpoints, expected result: %v", err)
+					}
+				}
```
Comment (maelk) [inline baremetal/cluster_manager_test.go]: this does not do anything if you do not get an error when you were expected one

File: baremetal/cluster_manager_test.go [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+				err = clusterMgr.UpdateClusterStatus()
+				if tc.ExpectSuccess {
+					if err != nil {
+						t.Error(err)
+					}
+				} else { // FAILURE is expected
+					if err != nil {
+						t.Logf("UpdateClusterStatus, expected result: %v", err)
+					}
+				}
```
Comment (maelk) [inline baremetal/cluster_manager_test.go]: same problem

File: baremetal/cluster_manager_test.go (hunk 1 above)
Comment (Jaakko-Os) [reply baremetal/cluster_manager_test.go]: used in 
func newSetup(t *testing.T, tc tcTest) (*ClusterManager, error)

I couldn't find another way to pass structure:

var testCases = map[string]struct {
	c             client.Client
	BMCluster     *infrav1.BareMetalCluster
	Cluster       *clusterv1.Cluster
	ExpectSuccess bool
}

to the newSetup function.

File: baremetal/cluster_manager_test.go (hunk 2 above)
Comment (Jaakko-Os) [reply baremetal/cluster_manager_test.go]: done

File: baremetal/cluster_manager_test.go (hunk 3 above)
Comment (Jaakko-Os) [reply baremetal/cluster_manager_test.go]: done

File: baremetal/cluster_manager_test.go (hunk 4 above)
Comment (Jaakko-Os) [reply baremetal/cluster_manager_test.go]: done

**Comment by kashifest** [comment]
looks good to me

**Pull request merged**